src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -572,6 +572,25 @@ public function text()
     }
 
     /**
+     * Returns the flattened node value of the first node of the list.
+     *
+     * Surrounding spaces/newlines are removed,
+     * and consecutive spaces/newlines are reduced to one space.
+     *
+     * @return string The flattened node value
+     *
+     * @throws \InvalidArgumentException When current node is empty
+     */
+    public function flatText()
+    {
+        if (!$this->nodes) {
+            throw new \InvalidArgumentException('The current node list is empty.');
```

Comment:
Throwing an exception is consistent with the rest of this class, which indeed I think is quite strict in practice (in my code I have to add a lot a tests for possibly absent nodes, instead of just getting null).